src/main/java/net/imagej/ops/image/invert/InvertIIInteger.java

```diff
@@ -52,83 +52,85 @@
 			final BigInteger minMax = minValue.add(maxValue);
 			mapper = Computers.unary(ops(), Ops.Map.class, output, input,
-				new AbstractUnaryComputerOp<I, O>()
+				new AbstractUnaryComputerOp<T, T>()
 			{
 
 					@Override
-					public void compute(I in, O out) {
-						BigInteger inverted = minMax.subtract(in.getBigInteger());	
+					public void compute(T in, T out) {
+						BigInteger inverted = minMax.subtract(((IntegerType)in).getBigInteger());	
 						
-						if( inverted.compareTo(minValue(out).getBigInteger()) <= 0) out.set(minValue(out));
-						else if(inverted.compareTo(maxValue(out).getBigInteger()) >= 0) out.set(maxValue(out));
-						else out.setBigInteger(inverted);
+						if( inverted.compareTo(((IntegerType)minValue(out)).getBigInteger()) <= 0) out.set(minValue(out));
+						else if(inverted.compareTo(((IntegerType)maxValue(out)).getBigInteger()) >= 0) out.set(maxValue(out));
+						else ((IntegerType)out).setBigInteger(inverted);
 						
 						
 					}
 				});
 		}
 		mapper.compute(input, output);
+
 	}
 
 	public static <T extends RealType<T>> T minValue(T type) {
-		if (type instanceof UnboundedIntegerType){
-			UnboundedIntegerType t = new UnboundedIntegerType();
-			t.setReal(0);
-			return (T) t;
-		}
-
-		T min = type.createVariable();
-		min.setReal(min.getMinValue());
-		return (T) min;
+		// TODO: Consider making minValue an op.
+		final T min = type.createVariable();
+		if (type instanceof UnboundedIntegerType) min.setReal(0);
+		else min.setReal(min.getMinValue());
+		return min;
 
 	}
-	
+
 	public static <T extends RealType<T>> T maxValue(T type) {
-		if (type instanceof Unsigned128BitType) {
-			Unsigned128BitType t = new Unsigned128BitType();
+		// TODO: Consider making maxValue an op.
+		final T max = type.createVariable();
+		if (max instanceof Unsigned128BitType) {
+			final Unsigned128BitType t = (Unsigned128BitType) max;
 			t.set(t.getMaxBigIntegerValue());
-			return (T) t;
 		}
-		else if (type instanceof UnsignedLongType) {
-			UnsignedLongType t = new UnsignedLongType();
+		else if (max instanceof UnsignedLongType) {
+			final UnsignedLongType t = (UnsignedLongType) max;
 			t.set(t.getMaxBigIntegerValue());
-			return (T) t;
 		}
-		else if (type instanceof UnboundedIntegerType) {
-			UnboundedIntegerType t = new UnboundedIntegerType();
-			t.setReal(0);
-			return (T) t;
-
+		else if (max instanceof UnboundedIntegerType) {
+			max.setReal(0);
 		}
-						
-			T t = type.createVariable();
-			t.setReal(type.getMaxValue());
-			return (T) t;
+		else {
+			max.setReal(type.getMaxValue());
+		}
+		return max;
+	}
 
+	@Override
+	public boolean conforms() {
+		final T inType = in().firstElement();
+		return inType instanceof Unsigned128BitType ||
+			inType instanceof UnsignedLongType ||
+			inType instanceof UnboundedIntegerType ||
+			inType instanceof LongType;
```

Comment:
Could this be replaced with something like:
```java
final T inType = in().firstElement();
if (inType instanceof IntegerType) {
	final IntegerType<?> copy = (IntegerType<?>) inType.createVariable();
	copy.setInteger(Long.MAX_VALUE);
	return copy.getIntegerLong() == Long.MAX_VALUE;
}
return false;
```

This way we don't need all the `instanceof`s (still one though 😿 ). And it would still work if another large `IntegerType` was added.

Reply:
Great!